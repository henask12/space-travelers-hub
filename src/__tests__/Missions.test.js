import React from 'react';
import { render } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect'; // For better DOM assertions
import MissionsList from '../components/Missions/MissionComp';

// Mocking useDispatch and useSelector
jest.mock('react-redux', () => ({
  useDispatch: () => jest.fn(),
  useSelector: (selectorFn) => selectorFn(
    { missions: { missions: [] } },
  ),
}));

test('renders the Empty Missions component correctly', () => {
  const { container } = render(<MissionsList />);
  expect(container).toMatchSnapshot();
});
