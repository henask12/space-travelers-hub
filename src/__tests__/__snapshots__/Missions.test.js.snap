// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the Empty Missions component correctly 1`] = `
<div>
  <div
    class="bg-white border border-gray-400 rounded-lg shadow-sm"
  >
    <div
      class="p-8"
    >
      <div
        class="bg-white border border-white-300 rounded-lg shadow-sm ml-8 mr-8"
      />
      <div
        class="flex flex-col"
      >
        <div
          class="-my-2 overflow-x-auto sm:-mx-6 lg:-mx-8"
        >
          <div
            class="py-2 align-middle inline-block min-w-full sm:px-6 lg:px-8"
          >
            <div
              class="shadow overflow-hidden border-b border-gray-200 sm:rounded-lg"
            >
              <table
                class="m-6 min-w-full divide-y divide-gray-200"
              >
                <thead
                  class="bg-gray-50"
                >
                  <tr>
                    <th
                      class="px-6 py-3 text-left text-xs font-bold text-black-500 uppercase tracking-wider"
                      scope="col"
                    >
                      Name
                    </th>
                    <th
                      class="px-6 py-3 text-left text-xs font-bold text-black-500 uppercase tracking-wider"
                      scope="col"
                    >
                      Title
                    </th>
                    <th
                      class="px-6 py-3 text-left text-xs font-bold text-black-500 uppercase tracking-wider"
                      scope="col"
                    >
                      Status
                    </th>
                    <th
                      class="px-6 py-3 text-left text-xs font-bold text-black-500 uppercase tracking-wider"
                      scope="col"
                    >
                      Join
                    </th>
                  </tr>
                </thead>
                <tbody
                  class="bg-white divide-y divide-gray-200"
                />
              </table>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
